### PR TITLE
Moved "Authorization settings" and "Application Settings" sub-modules from the UserAdmin app

### PR DESCRIPTION
**Pull request by un-tone**
For https://github.com/Starcounter/CompanyTrack/issues/245 and https://github.com/StarcounterPrefabs/UserAdmin/issues/58.
Was added UriMapping on /systemuser/authentication/settings/{?} for
Authentication Settings that will be used by UserAdmin app

### CODE REVIEW

File: src/SignIn/ViewModels/SettingsPage.json.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            {
+                return;
+            }
+
+            if (this.Transaction.IsDirty)
+            {
+                this.Transaction.Commit();
+            }
+
+            this.RedirectUrl = "/signin/profile";
```
Comment (warpech) [inline src/SignIn/ViewModels/SettingsPage.json.cs]: Pressing Save button on the Settings page should not redirect to `/signin/profile`. I think it should just give some message "Changes saved".

File: src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+using System.Web;
+using Simplified.Ring3;
+using Simplified.Ring6;
+using Smorgasbord.PropertyMetadata;
+using Starcounter;
+
+// FORGOT PASSWORD:
+// http://www.asp.net/identity/overview/features-api/account-confirmation-and-password-recovery-with-aspnet-identity
+
+namespace SignIn.ViewModels
```
Comment (warpech) [inline src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs]: In all our sample apps we have a convention that all classes are in a namespace `AppName`, so just `SignIn` (not `SignIn.ViewModels`).

What do you think about that? Should we change the convention?

File: src/SignIn/wwwroot/SignIn/viewmodels/SystemUserAuthenticationSettings.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            <sgb-property-metadata items="{{model.PropertyMetadataItems}}"></sgb-property-metadata>
+
+            <template is="dom-if" if="{{model.Message}}">
+                <p class="alert alert-danger">{{model.Message}}</p>
+            </template>
+
+            <h3>Authentication settings</h3>
+
+            <div>
+                <button class="btn btn-default btn-sm" onclick="++this.value;" value="{{model.NewPasswordClick$::click}}">Set New Password</button>
```
Comment (warpech) [inline src/SignIn/wwwroot/SignIn/viewmodels/SystemUserAuthenticationSettings.html]: For me this button does not work in Firefox. Perhaps change `onclick=` to `onmousedown=`, as in KitchenSink? https://github.com/StarcounterApps/KitchenSink/blob/master/src/KitchenSink/wwwroot/KitchenSink/ButtonPage.html#L22

This applies to all buttons

File: src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs (hunk 2 above)
Comment (joozek78) [inline src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs]: Yes! Definitely!

Comment (warpech) [inline src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs]: Moved: https://github.com/Starcounter/Guidelines/issues/14

File: src/SignIn/ViewModels/SettingsPage.json.cs (hunk 1 above)
Comment (un-tone) [inline src/SignIn/ViewModels/SettingsPage.json.cs]: Fixed.

File: src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs (hunk 2 above)
Comment (un-tone) [inline src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs]: For current case I replaced `SignIn.ViewModels` with `SignIn` to follow common style in scope of the project.

File: src/SignIn/wwwroot/SignIn/viewmodels/SystemUserAuthenticationSettings.html (hunk 3 above)
Comment (un-tone) [inline src/SignIn/wwwroot/SignIn/viewmodels/SystemUserAuthenticationSettings.html]: Replaced all `onlick` occurences with `onmousedown`

**Comment by un-tone** [comment]
The PR is updated. Please review.

File: src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs (hunk 2 above)
Comment (warpech) [inline src/SignIn/ViewModels/SystemUserAuthenticationSettings.json.cs]: Good approach, thanks

**Pull request merged**